packages/xrpl/test/models/CredentialAccept.test.ts

```diff
@@ -0,0 +1,153 @@
+import { stringToHex } from '@xrplf/isomorphic/dist/utils'
+import { assert } from 'chai'
+
+import { validate, ValidationError } from '../../src'
+import { validateCredentialAccept } from '../../src/models/transactions/CredentialAccept'
+
+/**
+ * AMMDelete Transaction Verification Testing.
+ *
+ * Providing runtime verification testing for each specific transaction type.
+ */
+describe('CredentialAccept', function () {
+  let credentialAccept
+
+  beforeEach(function () {
+    credentialAccept = {
+      TransactionType: 'CredentialAccept',
+      Issuer: 'r9LqNeG6qHxjeUocjvVki2XR35weJ9mZgQ',
+      Account: 'rNdY9XDnQ4Dr1EgefwU3CBRuAjt3sAutGg',
+      CredentialType: stringToHex('Passport'),
+      Sequence: 1337,
+      Flags: 0,
+    } as any
+  })
+
+  it(`verifies valid CredentialAccept`, function () {
+    assert.doesNotThrow(() => validateCredentialAccept(credentialAccept))
+    assert.doesNotThrow(() => validate(credentialAccept))
+  })
+
+  it(`throws w/ missing field Account`, function () {
+    delete credentialAccept.Issuer
+    const errorMessage = 'CredentialAccept: missing field Issuer'
+    assert.throws(
+      () => validateCredentialAccept(credentialAccept),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(
+      () => validate(credentialAccept),
+      ValidationError,
+      errorMessage,
+    )
+  })
+
+  it(`throws w/ Account not a string`, function () {
+    credentialAccept.Account = 123
+    const errorMessage = 'CredentialAccept: invalid field Account'
+    assert.throws(
+      () => validateCredentialAccept(credentialAccept),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(
+      () => validate(credentialAccept),
+      ValidationError,
+      errorMessage,
+    )
+  })
+
+  it(`throws w/ missing field Issuer`, function () {
+    delete credentialAccept.Issuer
```

Comment:
Since the Issuer field is mandatory in this transaction, is there a need to test this case? Typescript's interfaces should check if all the `required` fields are present.

Reply:
From my understanding, the tx parameter in the js library is not strongly typed, and instead uses a loose generic type definition. The actual fields each are manually checked, which is why the validate functions for all transactions make use of `validateRequiredField()`, which we have to manually define an error for. You can see this in other existing transactions like ammDeposit, which also has a test case for required fields like Asset, because it is possible to enter this part of the code missing a required field and instead run into other runtime errors.

I agree this seems avoidable if we restructure the repo to better use the typing (we already define the exact interface structure, why do we then manually check each required field instead of making use of those shapes?) and I raised this earlier, but that would impact all transactions so would be better as a different PR initiative to restructure the repo.

Reply:
okay, that makes sense.